src/Twilio/Security/RequestValidator.php

```diff
@@ -110,26 +110,11 @@ public function validate(string $expectedSignature, string $url, $data = []): bo
      * @return bool True if $a === $b, false otherwise.
      */
     public static function compare(?string $a, ?string $b): bool {
-        // if the strings are different lengths, obviously they're invalid
-        if (\strlen($a) !== \strlen($b)) {
-            return false;
+        if ($a && $b) {
+            return hash_equals($a, $b);
         }
 
-        if (!$a && !$b) {
-            return true;
-        }
-
-        $limit = \strlen($a);
-
-        // checking every character for an exact difference, if you find one, return false
-        for ($i = 0; $i < $limit; ++$i) {
-            if ($a[$i] !== $b[$i]) {
-                return false;
-            }
-        }
-
-        // there have been no differences found
-        return true;
+        return false;
```

Comment:
nit: If they are both empty, the previous implementation returns a true but this one returns a false?

Reply:
I don't think we'd actually ever get a situation where both arguments are null. (this is different from whether they're both empty strings, in which case this would return true as expected). In all of this function's uses, at least one of the parameters is computed. If anything, I think this function signature should change to not allow nullable arguments